arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/callTests/sixtyThreeSixtyFourths/SixtyThreeSixtyFourthsTests.java

```diff
@@ -0,0 +1,218 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class SixtyThreeSixtyFourthsTests {
+
+  /*
+  Cases to cover:
+
+  * value = 0
+  If precompileGasCost >= 2300 then we are interested in:
+   * value = 1, targetAddressExists = false
+   * value = 1, targetAddressExists = true
+
+  Note: BLAKE2F and MODEXP requires a non-zero non-trivial input to have a cost greater than 2300.
+  Other precompiles only require a proper call data size.
+  */
+
+  final Bytes gas = Bytes.fromHexString("ff".repeat(32));
+
+  // Generic program before the final call to the precompile when the target address does not exist
+  final BytecodeCompiler preCallTargetAddressDoesNotExistProgram =
+      BytecodeCompiler.newProgram()
+          .push(4096 - 32)
+          .op(OpCode.MLOAD)
+          .push(0) // returnAtCapacity
+          .push(0) // returnAtOffset
+          .push(0) // callDataSize
+          .push(0) // callDataOffset
+          .push(0) // value
+          .push(0) // address (this is 0 as the cost of PUSH is always the same)
+          .push(gas); // gas
+
+  // Cost of the generic program before the final call to the precompile when the target address
+  // does not exist
+  final long preCallTargetAddressDoesNotExistProgramGas =
+      BytecodeRunner.of(preCallTargetAddressDoesNotExistProgram).runOnlyForGasCost();
+
+  // Cost of the generic program before the final call to the precompile when the target address
+  // exists
+  Map<Address, Long> preCallTargetAddressExistsProgramGasMap = new HashMap<>();
+
+  @BeforeAll
+  void computePreCallTargetAddressExistsProgramGasOnceForAll() {
+    // Generic program before the final call to the precompile when the target address exists
+    final Function<Address, BytecodeCompiler> preCallTargetAddressExistsProgram =
+        (address) ->
+            BytecodeCompiler.newProgram()
+                .push(0) // returnAtCapacity
+                .push(0) // returnAtOffset
+                .push(0) // callDataSize
+                .push(0) // callDataOffset
+                .push(0) // value
+                .push(address) // address
+                .push(gas) // gas
+                .op(OpCode.CALL)
```

Comment:
I would cut this in three. There is no relation between these three operations. Also if you want the `address` to exist you need to transfer a **nonzero** value.